src/SignalR/server/Core/src/Internal/HubMethodDescriptor.cs

```diff
@@ -83,7 +84,11 @@ public HubMethodDescriptor(ObjectMethodExecutor methodExecutor, IServiceProvider
                 }
                 else if (serviceProviderIsService?.IsService(p.ParameterType) == true)
                 {
-                    Debug.Assert(index < 32);
+                    if (index >= 32)
```

Comment:
Doesn't this feel bad? Maybe just allocate an array of bools after 32.

Reply:
No one should be doing this, it's just a nicer error than what you would get if we didn't have a check. We could change to long and support 64 params if you really want

Reply:
I agree with you that nobody should ever hit this, until we hit that person that codegenerated hub methods...